### ISSUE TITLE
[1.12] Crash on single player client during exploring nether

### ISSUE THREAD

**Issue by MechaJ** [MightyPirates/BedrockOres#17]
```
WARNING: coremods are present:
  LibLoader (# LibLoader.jar)
  CXLibraryCore (cxlibrary-1.12-1.5.1.jar)
  ForgelinPlugin (Forgelin-1.5.0.jar)
  TANLoadingPlugin (ToughAsNails-1.12-3.0.0.62-universal.jar)
  LoadingPlugin (HardcoreDarkness-MC1.12-1.9.jar)
  LoadingPlugin (RandomThings-MC1.12.1-4.0.5.jar)
  LoadingPlugin (Bloodmoon-MC1.12-1.5.1.jar)
  ShetiPhian-ASM (shetiphiancore-1.12.0-3.5.0.2.jar)
  Do not report to Forge! Remove FoamFixAPI (or replace with FoamFixAPI-Lawful) and try again. (foamfix-0.8.0-beta2-1.12-anarchy.jar)
  CTMCorePlugin (CTM-MC1.12-0.2.2.7.jar)
  IvToolkit (IvToolkit-1.3.3-1.12.jar)
  IMLoadingPlugin (infinimend-1.12-1.0.jar)
  AstralCore (astralsorcery-1.12-1.6.1.jar)
  LoadingPlugin (Quark-r1.2-103.jar)
  Smart HUD Core (smarthud-1.12-1.0.4.jar)
  IELoadingPlugin (ImmersiveEngineering-0.12-71.jar)
  MC64836CoreMod (mc64836-0.1.1.jar)
  DLFMLCorePlugin (DynamicLights-1.12.jar)
  HammerCoreCore (HammerCore-1.12-1.9.4.0.jar)
  CoreMod (TickProfiler-1.12-jenkins.15.jar)
  FullscreenFix (gud_fullscreen_fix-1.0.2.jar)
  BlurPlugin (Blur-1.0.3-13.jar)
  CreativePatchingLoader (CreativeCore v1.8.7 mc1.12.1.jar)
  AppleCore (AppleCore-mc1.12-2.2.2.jar)
  Plugin (NotEnoughIDs-1.5.4.1.jar)
Contact their authors BEFORE contacting forge

// I just don't know what went wrong :(

Time: 8/27/17 8:24 PM
Description: Exception ticking world

java.lang.ArrayIndexOutOfBoundsException: Array index out of range: 1
	at gnu.trove.list.array.TFloatArrayList.get(TFloatArrayList.java:281)
	at li.cil.bedrockores.common.world.WorldGeneratorBedrockOre.generateImpl(WorldGeneratorBedrockOre.java:169)
	at li.cil.bedrockores.common.world.WorldGeneratorBedrockOre.generate(WorldGeneratorBedrockOre.java:39)
	at net.minecraftforge.fml.common.registry.GameRegistry.generateWorld(GameRegistry.java:167)
	at net.minecraft.world.chunk.Chunk.func_186034_a(Chunk.java:1015)
	at net.minecraft.world.chunk.Chunk.func_186030_a(Chunk.java:981)
	at net.minecraft.world.gen.ChunkProviderServer.func_186025_d(ChunkProviderServer.java:157)
	at net.minecraft.world.World.func_72964_e(World.java:308)
	at net.minecraft.world.World.func_175726_f(World.java:303)
	at net.minecraft.world.World.func_180495_p(World.java:908)
	at net.minecraft.world.chunk.Chunk.func_76594_o(Chunk.java:1272)
	at net.minecraft.world.WorldServer.func_147456_g(WorldServer.java:400)
	at net.minecraft.world.WorldServer.func_72835_b(WorldServer.java:224)
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:750)
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:665)
	at net.minecraft.server.integrated.IntegratedServer.func_71217_p(IntegratedServer.java:185)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:523)
	at java.lang.Thread.run(Thread.java:745)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Client thread
Stacktrace:
	at gnu.trove.list.array.TFloatArrayList.get(TFloatArrayList.java:281)
	at li.cil.bedrockores.common.world.WorldGeneratorBedrockOre.generateImpl(WorldGeneratorBedrockOre.java:169)
	at li.cil.bedrockores.common.world.WorldGeneratorBedrockOre.generate(WorldGeneratorBedrockOre.java:39)
	at net.minecraftforge.fml.common.registry.GameRegistry.generateWorld(GameRegistry.java:167)
	at net.minecraft.world.chunk.Chunk.func_186034_a(Chunk.java:1015)
	at net.minecraft.world.chunk.Chunk.func_186030_a(Chunk.java:981)
	at net.minecraft.world.gen.ChunkProviderServer.func_186025_d(ChunkProviderServer.java:157)
	at net.minecraft.world.World.func_72964_e(World.java:308)
	at net.minecraft.world.World.func_175726_f(World.java:303)
	at net.minecraft.world.World.func_180495_p(World.java:908)
	at net.minecraft.world.chunk.Chunk.func_76594_o(Chunk.java:1272)
	at net.minecraft.world.WorldServer.func_147456_g(WorldServer.java:400)
	at net.minecraft.world.WorldServer.func_72835_b(WorldServer.java:224)

-- Affected level --
Details:
	Level name: New World
	All players: 1 total; [EntityPlayerMP['MechaJ'/355, l='New World', x=534.52, y=93.74, z=-687.48]]
	Chunk stats: ServerChunkCache: 2135 Drop: 0
	Level seed: 2094096012
	Level generator: ID 00 - default, ver 1. Features enabled: false
	Level generator options: 
	Level spawn location: World: (0,0,0), Chunk: (at 0,0,0 in 0,0; contains blocks 0,0,0 to 15,255,15), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,0,0 to 511,255,511)
	Level time: 0 game time, 0 day time
	Level dimension: 0
	Level storage version: 0x00000 - Unknown?
	Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
	Level game mode: ~~ERROR~~ NullPointerException: null
Stacktrace:
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:750)
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:665)
	at net.minecraft.server.integrated.IntegratedServer.func_71217_p(IntegratedServer.java:185)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:523)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by fnuecke** [comment]
That... should be impossible now :/

What version of the mod does this happen with? If it's pre-MC1.12-1.0.0.12 please try updating and see if that helps.

**Comment by MechaJ** [comment]
Bedrock Ores-MC1.12-1.0.2.16.jar

**Comment by fnuecke** [comment]
Hmm, actually this *might* happen if the world gen triggers further chunks to gen. Which it shouldn't. But I'm trying to doubly enforce that it doesn't now; should be fixed in version 1.0.3, either wait for Curse to approve it or grab it from [the build server](http://ci.cil.li/job/BedrockOres-MC1.12/).

Thanks for the report btw!

**Comment by MechaJ** [comment]
The bedrock ore isn't being generated at all with that build.  And thank you for the most excellent mod!

**Comment by fnuecke** [comment]
Wat. (And thanks :x)
I'll have to have a look in the evening, trying to patch things via Github's webinterface doesn't seem to work so well >_>

**Comment by fnuecke** [comment]
Allright, fixed in 1.0.4!
